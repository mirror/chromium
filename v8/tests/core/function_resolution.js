function f1() {
  print("outer f1");
}


function f0() {
  f1();
  f1 = function() {
    print("assigned f1");
  }
  f1();
  function f1() {
    print("declared f1");
  }
  f1();
}


f0();


function f2() {
  function g0() {
    print("g0");
  }
  var g0;  // variable declaration must leave g0 untouched
  g0();
}


f2();
