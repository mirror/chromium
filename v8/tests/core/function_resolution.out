declared f1
assigned f1
assigned f1
g0
